### ISSUE TITLE
Uploading file via API does not stage files in git

### ISSUE THREAD

**Issue by ionmincu** [WeblateOrg/weblate#2416]
**Describe the bug**
Uploading a resx file with the API (updating an existing resx component) will set the file as modified but it won't be in the staging area, making the `Commit` button from the UI useless (pressing the commit button won't actually commit the changes, they will just remain as modified files).

**To Reproduce**
1. Create project with a resx component synced from git
2. Modify the resx file locally
3. Upload the updated resx file via the API using the url `/api/translations/(string: project)/(string: component)/(string: language)/file/` with `overwrite` set to `true`
4. Press commit
5. In weblate you will see pending changes, going into the `/app/data/vcs/{project}/{component}`  folder and type `git status`

**Expected behavior**
Expected message from `git status` command is:
```
On branch MY_WEBLATE_BRANCH
Your branch is up-to-date with 'origin/MY_WEBLATE_BRANCH'.
nothing to commit, working tree clean
```

**Actual behavior**
Actual  `git status` message is:
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   SOME_FOLDER/SOME_FILE.resx

no changes added to commit (use "git add" and/or "git commit -a")
```

**PLEASE NOTE:**
Sometimes this actually works, the files will be commited.

**Server configuration and status**
```
 * Weblate 3.1.1
 * Python 3.5.3
 * Django 2.0.8
 * six 1.10.0
 * social-auth-core 1.7.0
 * social-auth-app-django 2.1.0
 * django-appconf 1.0.2
 * translate-toolkit 2.3.0
 * Whoosh 2.7.4
 * defusedxml 0.5.0
 * Git 2.11.0
 * Pillow 4.0.0
 * python-dateutil 2.5.3
 * lxml 3.7.1
 * django-crispy-forms 1.7.2
 * django_compressor 2.2
 * djangorestframework 3.8.2
 * user-agents 1.1.0
 * jellyfish 0.6.1
 * pytz 2018.5
 * pyuca 1.2
 * PyYAML 3.12
 * tesserocr 2.3.0
 * Mercurial 4.0
 * git-svn 2.11.0
 * Database backends: django.db.backends.postgresql
 * Cache backends: avatar:FileBasedCache, default:MemcachedCache
 * Platform: Linux 4.15.0-1023-azure (x86_64)
```

**Additional context**
Running from docker container

**Comment by nijel** [comment]
This should be addressed in 3.2 series - the upload now behave same as translating, it used to have custom code. Can you please try to reproduce with 3.2? (Or with 3.3 which will be released tomorrow).

**Comment by ionmincu** [comment]
Ok, I will upgrade sometimes in the future, till then I think you can close the issue, I will adress it if it happens on 3.3.
